enhancements/evidence-forwarding.md

```diff
@@ -0,0 +1,305 @@
+
+### Goals
+
+The goal is to provide a way for Kubernetes administrators to configure the
+Compliance Operator to send compliance reports, specifically ARF reports, to an
+external storage pool, like an object or file store.
+
+### Non-Goals
+
+It is not the goal of this enhancement to implement results forwarding, or
+adjust the definition of evidence outside an ARF report.
+
+## Proposal
+
+Provide an additional configuration for scan settings that allows site
+reliability engineers to forward evidence to a storage pool. Today, the
+Compliance Operator provides a `default` `ScanSetting` that integrates with
+persistent storage:
+
+```yaml
+---
+apiVersion: compliance.openshift.io/v1alpha1
+kind: ScanSetting
+metadata:
+  name: default
+  namespace: openshift-compliance
+rawResultStorage:
+  pvAccessModes:
+  - ReadWriteOnce
+  rotation: 3
+  size: 1Gi
+roles:
+- worker
+- master
+scanTolerations:
+  default:
+  - operator: Exists
+  schedule: 0 1 * * *
+```
+
+A site reliability engineer could create another `ScanSetting` that bypasses
+storing evidence on physical volumes with the following:
+
+```yaml
+---
+apiVersion: compliance.openshift.io/v1alpha1
+kind: ScanSetting
+metadata:
+  name: external-storage
+  namespace: openshift-compliance
+externalResultStorage:
+  endpoint: https://object-storage.example.com/
+  tls:
+roles:
+- worker
+- master
+scanTolerations:
+  default:
+  - operator: Exists
+  schedule: 0 1 * * *
+```
+
+A site reliability engineer could configure both storage mechanisms for redundancy:
+
+```yaml
+---
+apiVersion: compliance.openshift.io/v1alpha1
+kind: ScanSetting
+metadata:
+  name: external-storage-and-cluster-storage
+  namespace: openshift-compliance
+rawResultStorage:
+  pvAccessModes:
+  - ReadWriteOnce
+  rotation: 3
+  size: 1Gi
+externalResultStorage:
+  endpoint: https://object-storage.example.com/
+  tls:
+roles:
+- worker
+- master
+scanTolerations:
+  default:
+  - operator: Exists
+  schedule: 0 1 * * *
+```
+
+### User Stories
+
+1. As a site reliability engineer, I want to configure the Compliance Operator
+   to send all ARF reports to a dedicated object store.
+2. As a site reliability engineer, I want to opt out of storing any ARF reports
+   on physical volumes.
+3. As a site reliability engineer, I expect verbose alerts when the Compliance
+   Operator cannot send ARF reports to a configured external storage pool.
+4. As a site reliability engineer, I expect the Compliance Operator to fail if
+   the configured endpoint is unreachable.
+5. As a site reliability engineer, I expect to forward evidence to
+   public object-storage endpoints, like AWS S3.
+5. As a site reliability engineer, I expect to forward evidence to
+   private object-storage endpoints in cluster, Ceph or OpenShift Data Foundation.
+
+### API Extensions
+
+This proposal would extend the `ScanSetting` custom resource definition (CRD)
+to include an additional attribute called `externalResultStorage`.
```

Comment:
To my point about the naming wrt evidence, maybe this can be `externalRawResultStorage` (or `externalEvidenceStorage` if we stick with it)

Reply:
I like the idea of using `externalRawResultStorage` and `externalResultStorage`. We could also make it implementation specific (e..g, `s3RawResultStorage` and `odfRawResultStorage`).

Additionally, we could use `grpcResultStorage`. Thoughts?